common/src/main/java/com/wynntils/models/trademarket/TradeMarketModel.java

```diff
@@ -72,4 +85,32 @@ public void setPresetFilter(int presetId, String filter) {
         presetFilters.get().put(presetId, filter);
         presetFilters.touched();
     }
+
+    public TradeMarketPriceInfo calculateItemPriceInfo(ItemStack itemStack) {
+        StyledText priceLine = LoreUtils.getLoreLine(itemStack, TRADE_MARKET_PRICE_LINE);
+
+        if (priceLine == null || !priceLine.matches(PRICE_STR)) {
+            WynntilsMod.warn("Trade Market item had an unexpected price line: " + priceLine);
```

Comment:
This warning might just as well mean "we're not in the trade market"..? Or is there some mechanism I am missing that means we should only end up here for items on the TM?

Reply:
Theoretically this code supports calculating price in other guis from any other models/features, if needed

Reply:
Hm, what I mean is that do we not risk spamming the console with "Trade Market item had an unexpected price line: ..." if we try to access price stats outside the TM? As in, maybe we should not print a warning but just return.

If the first line match but the second does not, by all means -- print a warning! Then the price info has changed and our regexp fails. But if there is no "Price"-line, then it is not really an error.

Reply:
Hm, I see what you mean. Code that is only supposed to run in the trade market (hence why it is in the TM model) got mixed up with some general purpose stuff... I can remove the log if you want, maybe not the best solution though.

Reply:
I think it is an okay solution to just remove the log. If you try to filter your items on price outside TM, you will have to take the hit that we try to read the first line of the lore of all items in the container, just to be able to return an EMPTY price info.

I take it the stat providers are not actively doing anything unless the user explicitly tries to filter on them?

Reply:
> I take it the stat providers are not actively doing anything unless the user explicitly tries to filter on them?

Yes, that is the case. These are only parsed when they are filtered for, on items that are already filtered.

Reply:
Hm, wait a bit! I just read the code, and seems like I removed all "general" use case scenarios from it, and only works in the TM. That means that every TM item is expected to have that proper lore, so I think the warning is appropriate in both cases.

Reply:
Ok, if this is the case, then the code is fine.